pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -382,7 +383,10 @@ func (r *ReconcileCredentialsRequest) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, fmt.Errorf("configuration conflict")
 	} else if mode == operatorv1.CloudCredentialsModeManual {
 		logger.Infof("operator set to disabled / manual mode")
-		return reconcile.Result{}, err
+		if !stsDetected {
+			logger.Infof("operator detects STS enabled cluster")
+			return reconcile.Result{}, err
```

Comment:
This return short circuits continuing through the controller to create the secret for the CredentialsRequest.

Reply:
Worth adding a test to the credentials request controller tests to ensure we go through the motions.

Reply:
It's the end of the day and it's been a long one for me, again, I'm not seeing a good way to create a test on the controller with STS enabled. There might be one, I'm just kind of lost at the moment. Maybe we should add a card to Jira for "Streamline Credentials Request Controller Unit Tests and add STS enabled and feature gate flags"?